build-library/action.yml

```diff
@@ -0,0 +1,54 @@
+name: "Build a Python library"
+description: "Build a Python library by generating wheels and sdist artifacts"
+
+inputs:
+  library-name:
+    description: 'Name of the Python library'
+    required: true
+    type: string
+  python-version:
+    description: 'Desired Python version for pre-commit'
+    default: '3.10'
+    required: false
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+  
+    - name: "Install Git and clone project"
+      uses: actions/checkout@v3
+  
+    - name: "Setup Python"
+      uses: actions/setup-python@v4.2.0
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: "Check if requirements_build.txt file exists"
+      shell: bash
+      run: |
+        echo "EXISTS_BUILD_REQUIREMENTS=$(if [ -f requirements/requirements_build.txt ]; then echo 'true'; else echo 'false'; fi)" >> $GITHUB_ENV
+
+    - name: "Install documentation dependencies from requirements file"
+      if: env.EXISTS_BUILD_REQUIREMENTS == 'true'
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install -r requirements/requirements_build.txt
+
+    - name: "Install build dependencies from pyproject.toml"
+      if: env.EXISTS_BUILD_REQUIREMENTS == 'false' 
+      shell: bash 
+      run: python -m pip install .[build]
```

Comment:
Interesting... all projects now have to define the same targets?...

For example, in PyGeometry we only have docs and test extra targets. We don't have a "build" target.

Reply:
Yeah, this is something I have been thinking.

In the end, any Python library should be built using [build] and [twine], as these are the official libraries maintained by PyPA. Any build system should be compliant with those.

Maybe we could get rid of these checks and just impose:

```bash
python -m pip install build twine
python -m build && python -m twine check
```

Reply:
I agree. That would be the best default option I think =)